### PR TITLE
Add checkout/checkin functionality to common_utils to lock access to S3 files

### PR DESCRIPTION
**Pull request by davidparks21**
Three functions added to common_utils.py: checkout, checkin, and force_release_checkout. These files read and write a file on S3 using smart_open and allow atomic access to that file (assuming everyone uses checkout/checkin correctly) using the MessageBroker's get_lock feature.

### CODE REVIEW

File: src/braingeneers/utils/common_utils.py
Comment (DailyDreaming) [inline src/braingeneers/utils/common_utils.py:222]: We should make a TODO to fix this.

File: src/braingeneers/utils/common_utils_test.py
Comment (DailyDreaming) [reply src/braingeneers/utils/common_utils_test.py:3]: Also evidence that this test is not running.

**Review by DailyDreaming — CHANGES_REQUESTED** [review]
I like this overal, and the test looks good to me, but I don't believe our tests are running in CI/CD.

Checkout this output: https://github.com/braingeneers/braingeneerspy/actions/runs/8024398343/job/21922894456?pr=68

We're only running files in the [tests directory](https://github.com/braingeneers/braingeneerspy/tree/master/tests).  So we should place all of the test files there.

I'm disinclined to use `global`s and in-function imports to get around circular imports.  I think it might take a broad refactor at this point to correct however.  Will message in Slack.

The tests should be fixed (and run in CI) before merging this I think.

File: src/braingeneers/utils/common_utils.py
Comment (davidparks21) [reply src/braingeneers/utils/common_utils.py:269]: This issue comes back to whether we should make dependencies for certain sub packages optional. For example the `ml` subpackage or `analysis` subpackage require some big dependencies like PyTorch and Matplotlib which are really hard to install and a raspberry PI. 

If we make sub packages optional, then we can't import across packages up at the top, otherwise some functions in common_utils.py wouldn't be accessible to a Raspberry PI for example. Maybe we can chat about this Friday on the Software meeting.

File: src/braingeneers/utils/common_utils_test.py
Comment (davidparks21) [reply src/braingeneers/utils/common_utils_test.py:3]: Yeah, we should fix that. I'd leave that to you if it's ok, but I would suggest putting that on your backlog behind the workflow tasks. Automated unit tests are quite valuable long run, for sure, but we have a number of people backlogged on the workflow automation at the moment, so that's higher priority. I'll fix this before re-submitting, but I think we can merge this without the automated unit tests.

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thanks for adding the tests.  I removed some unused imports and added some typing hints that were missing.  I also tested it out and it works as advertised.  This looks good to merge to me.

**Pull request merged**
